File: LineSDKStarterObjC/LineSDKStarterObjC/ViewController.m
//
//  ViewController.m
//  LineSDKStarterObjC
//
//  Created by Serrano Mark on 12/8/16.
//  Copyright © 2016 LINE. All rights reserved.
//

#import "ViewController.h"
#import "UserInfoViewController.h"
#import <LineSDK/LineSDK.h>

@interface ViewController () <LineSDKLoginDelegate>

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    // Set the LINE Login Delegate
    [LineSDKLogin sharedInstance].delegate = self;
}


- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)A2ALogin
{
    [[LineSDKLogin sharedInstance] startLoginWithPermissions:@[@"profile", @"openid"]];
}

- (IBAction)WebLogin
{
    [[LineSDKLogin sharedInstance] startWebLoginWithPermissions:@[@"profile", @"openid"]];
}


#pragma mark LineSDKLoginDelegate

- (void)didLogin:(LineSDKLogin *)login
      credential:(LineSDKCredential *)credential
         profile:(LineSDKProfile *)profile
           error:(NSError *)error
{
    if (error) {
        NSLog(@"LINE Login Failed with Error: %@", error.description);
        return;
    }
    
    NSLog(@"LINE Login Succeeded");
    
    UserInfoViewController *userInfoVC = [self.storyboard instantiateViewControllerWithIdentifier:@"userInfoViewController"];
    
    NSMutableDictionary *profileData = [@{
                                   @"userid" : profile.userID,
                                   @"displayname" : profile.displayName,
                                   @"accesstoken" : credential.accessToken.accessToken
                                   } mutableCopy];
    
    if(profile.pictureURL != nil) {
        profileData[@"pictureurl"] = profile.pictureURL;
    }
    
    if(profile.statusMessage != nil) {
        profileData[@"statusmessage"] = profile.statusMessage;
    }
    
    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
    [formatter setDateFormat:@"dd-MM-yyyy HH:mm"];

    NSMutableDictionary *openIDData = [@{
                                         @"issuer" : credential.IDToken.issuer,
                                         @"subject" : credential.IDToken.subject,
                                         @"audience" : credential.IDToken.audience,
                                         @"expiration" : [formatter stringFromDate:credential.IDToken.expiration],
                                         @"issueAt" : [formatter stringFromDate:credential.IDToken.issueAt],
                                         @"name" : credential.IDToken.name,
                                         @"pictureUrl" : credential.IDToken.pictureURL.absoluteString
                                         }mutableCopy];
    
    
    // Pass the user information into the next view controller so that we can display it.
    userInfoVC.profileData = profileData;
    userInfoVC.openIDData = openIDData;
    
    [self presentViewController:userInfoVC animated:YES completion:nil];
}

@end

File: LineSDKStarterSwift/LineSDKStarterSwift/ViewController.swift
//
//  ViewController.swift
//  LineSDKStarterSwift
//
//  Created by Serrano Mark on 2017/02/27.
//  Copyright © 2017 LINE. All rights reserved.
//

import UIKit
import LineSDK

class ViewController: UIViewController, LineSDKLoginDelegate {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        LineSDKLogin.sharedInstance().delegate = self
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func A2ALogin (with sender: UIButton) {
        
        LineSDKLogin.sharedInstance().start()
    }
    
    @IBAction func WebLogin (with sender: UIButton) {
        
        LineSDKLogin.sharedInstance().startWebLogin()
    }
    
    // MARK: LineSDKLoginDelegate
    
    func didLogin(_ login: LineSDKLogin, credential: LineSDKCredential?, profile: LineSDKProfile?, error: Error?) {
        
        if let error = error {
            print("LINE Login Failed with Error: \(error.localizedDescription) ")
            return
        }
        
        guard let profile = profile, let credential = credential, let accessToken = credential.accessToken else {
            print("Invalid Repsonse")
            return
        }
        
        print("LINE Login Succeeded")
        print("Access Token: \(accessToken.accessToken)")
        print("User ID: \(profile.userID)")
        print("Display Name: \(profile.displayName)")
        print("Picture URL: \(profile.pictureURL as URL?)")
        print("Status Message: \(profile.statusMessage as String?)")
        
        if let userInfoVC = self.storyboard?.instantiateViewController(withIdentifier: "userInfoViewController") as? UserInfoViewController {
            var data = ["userid" : profile.userID,
                        "displayname" : profile.displayName,
                        "accesstoken" : accessToken.accessToken]
            
            if let pictureURL = profile.pictureURL {
                data["pictureurl"] = pictureURL.absoluteString
            }
            
            if let statusMessage = profile.statusMessage {
                data["statusmessage"] = statusMessage
            }
            
            userInfoVC.userData = data
            self.present(userInfoVC, animated: true, completion: nil)
        }
    }
}
